src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/ApplicationTest.kt

```diff
@@ -1872,295 +1874,426 @@ class ApplicationTest : IntegrationTestBase() {
+      assertThat(persistedDomainEvents).hasSize(2)
+      val persistedApplicationSubmittedEvent = persistedDomainEvents.firstOrNull { it.type == DomainEventType.APPROVED_PREMISES_APPLICATION_SUBMITTED }
+      val persistedAssessmentAllocatedEvent = persistedDomainEvents.firstOrNull { it.type == DomainEventType.APPROVED_PREMISES_ASSESSMENT_ALLOCATED }
 
-            val persistedApplication = approvedPremisesApplicationRepository.findByIdOrNull(applicationId)!!
+      assertThat(persistedApplicationSubmittedEvent).isNotNull
+      assertThat(persistedApplicationSubmittedEvent!!.crn).isEqualTo(offenderDetails.otherIds.crn)
 
-            assertThat(persistedApplication.isWomensApplication).isFalse()
-            assertThat(persistedApplication.isPipeApplication).isTrue
-            assertThat(persistedApplication.targetLocation).isEqualTo("SW1A 1AA")
-            assertThat(persistedApplication.sentenceType).isEqualTo(SentenceTypeOption.nonStatutory.toString())
-            assertThat(persistedApplication.apArea?.id).isEqualTo(submittingUser.apArea!!.id)
+      assertThat(persistedAssessmentAllocatedEvent).isNotNull
+      assertThat(persistedAssessmentAllocatedEvent!!.crn).isEqualTo(offenderDetails.otherIds.crn)
 
-            val createdAssessment =
-              approvedPremisesAssessmentRepository.findAll().first { it.application.id == applicationId }
-            assertThat(createdAssessment.allocatedToUser!!.id).isEqualTo(assessorUser.id)
-            assertThat(createdAssessment.createdFromAppeal).isFalse()
+      domainEventAsserter.blockForEmittedDomainEvent(DomainEventType.APPROVED_PREMISES_ASSESSMENT_ALLOCATED)
 
-            val persistedDomainEvents = domainEventRepository.findAll().filter { it.applicationId == applicationId }
-            val persistedApplicationSubmittedEvent = persistedDomainEvents.firstOrNull { it.type == DomainEventType.APPROVED_PREMISES_APPLICATION_SUBMITTED }
-            val persistedAssessmentAllocatedEvent = persistedDomainEvents.firstOrNull { it.type == DomainEventType.APPROVED_PREMISES_ASSESSMENT_ALLOCATED }
+      emailAsserter.assertEmailsRequestedCount(2)
+      emailAsserter.assertEmailRequested(
+        toEmailAddress = createdAssessment.allocatedToUser!!.email!!,
+        templateId = notifyConfig.templates.assessmentAllocated,
+        replyToEmailId = persistedApplication.apArea!!.notifyReplyToEmailId,
+      )
+      emailAsserter.assertEmailRequested(
+        toEmailAddress = submittingUser.email!!,
+        templateId = notifyConfig.templates.applicationSubmitted,
+        replyToEmailId = persistedApplication.apArea!!.notifyReplyToEmailId,
+      )
+    }
 
-            assertThat(persistedApplicationSubmittedEvent).isNotNull
-            assertThat(persistedApplicationSubmittedEvent!!.crn).isEqualTo(offenderDetails.otherIds.crn)
+    @Test
+    fun `Submit short notice application auto allocates the assessment according to overridden ap area`() {
```

Comment:
Possibly change test name to 'Submit short notice application auto allocates the assessment according to overridden ap area and sends emails'()

Reply:
good spot